syntax = "proto3";

package com.webank.eggroll.core.meta;

import "meta.proto";

message StartContainersRequest {
  string session_id = 1;
  string name = 2;
  string job_type = 3;
  uint32 world_size = 4;
  repeated string command_arguments = 5;
  map<string, string> environment_variables = 6;
  map<string, bytes> files = 7;
  map<string, bytes> zipped_files = 8;
  map<string, string> options = 9;
  string status = 10;
  repeated Processor processors = 11;
}

message StartContainersResponse {
  string session_id = 1;
}

message StopContainersRequest {
  string session_id = 1;
}

message StopContainersResponse {
}

message KillContainersRequest {
  string session_id = 1;
}

message KillContainersResponse {
}

message DownloadContainersRequest {
  string session_id = 1;
  repeated int64 container_ids = 2;
  string compress_method = 3;
}

message DownloadContainersResponse {
  string session_id = 1;
  repeated ContainerContent container_content = 2;
}

message ContainerContent {
  int64 container_id = 1;
  bytes content = 2;
  string compress_method = 3;
}